ibtk/src/boundary/HierarchyGhostCellInterpolation.cpp

```diff
@@ -334,9 +334,7 @@ HierarchyGhostCellInterpolation::initializeOperatorState(
 
         const std::vector<RobinBcCoefStrategy<NDIM>*>& robin_bc_coefs = d_transaction_comps[comp_idx].d_robin_bc_coefs;
         bool null_bc_coefs = true;
-        for (std::vector<RobinBcCoefStrategy<NDIM>*>::const_iterator cit = robin_bc_coefs.begin();
-             cit != robin_bc_coefs.end();
-             ++cit)
+        for (auto cit = robin_bc_coefs.begin(); cit != robin_bc_coefs.end(); ++cit)
```

Comment:
I have a slight preference for writing
```cpp
for (const RobinBcCoefStrategy<NDIM> * cit : robin_bc_coefs)
```
here, but I may be outvoted :)

Won't the for-loop modernization redo all of these anyway?

Reply:
Yes we should be using range-based for.

Should switch to `auto` and range-based for at the same time?

Reply:
These gargantuan patches are hard enough; I think it would be easier to do it in sequence.

Reply:
Although it probably could/should be
```
for (const RobinBcCoefStrategy<NDIM> * bc_strategy : robin_bc_coefs)
```
since it is no longer an iterator. (FWIW, `it` and `cit` are used pretty consistently in the codebase to indicate iterators and `const`-iterators, respectively.)

Reply:
I like being able to see types at a glance so I personally do not write `auto` in ranged-for loops. Its been awhile since I tried, but IIRC all IDEs I tried cannot figure out the type used here either (I should try again before we put much weight in that).